### PR TITLE
Resolving the bug-fix

### LINKED ISSUE #2285: API fails if the `SERVER_PORT` variable is not in the `.env` file
**Describe the bug**

API fails if the `SERVER_PORT` variable is not in the `.env` file

**To Reproduce**

1. Remove the SERVER_PORT variable from the `.env` file
2. Start the API
3. Errors are seen

**Expected behavior**

1. The API should default to a SERVER_PORT if the variable is not in the `.env` file
2. We need valid passing tests created to make sure that this is not repeated

**Actual behavior**

- See above

**Screenshots**

image

**Additional details**

The feature was added in this PR, but this edge case was not considered

- https://github.com/PalisadoesFoundation/talawa-api/issues/2259

**Potential internship candidates**

Please read this if you are planning to apply for a Palisadoes Foundation internship https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by Suyash878**
**Issue Number:**

Fixes #2285

**Did you add tests for your changes?**

Not relevant

**Snapshots/Videos:**

None

**If relevant, did you update the documentation?**

Not relevant

**Summary**

This PR is in reference to a previously closed PR #2377.

**Does this PR introduce a breaking change?**

**Other information**

I highly apologize for the repetition and inconsistency of my previous few PRs in my last PR I made a mistake and merged the main branch into my own newly created branch, I tried reset using the commit hash but I couldn't find the commit hash locally in my terminal.
I deeply regret any inconvenience this may have caused and am committed to resolving these issues promptly.

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-api/blob/master/CONTRIBUTING.md)?**

Yes

## Summary by CodeRabbit

- **Bug Fixes**
  - Improved server initialization to dynamically capture and set the server host and port.
  - Enhanced error handling for server port validation.

- **New Features**
  - Server and subscription endpoint URLs are now dynamically adjusted based on the request host, ensuring correct API root URL setup.

### CODE REVIEW

**Comment by Suyash878** [comment]
@palisadoes Kindly review

**Comment by varshith257** [comment]
@Suyash878 Better add a video to get ease for review of your changes

**Comment by Suyash878** [comment]
[Screencast from 2024-07-11 13-31-30.webm](https://github.com/PalisadoesFoundation/talawa-api/assets/141909257/c088feaa-e197-4c60-ae44-c3412c43fc97)
Here's a video for better understanding.

**Comment by palisadoes** [comment]
1. What happens when the .env file doesn't have the parameter? 
2. Does the error in your video appear then? 
3. If so, why? Ideally, we need to protect against that.

File: src/index.ts [hunk 1]
```diff
@@ -110,31 +111,37 @@ async function startServer(): Promise<void> {
...
+        };
+      },
     }),
   );
 
   // Modified server startup
-  await new Promise<void>((resolve) =>
-    httpServer.listen({ port: parseInt(SERVER_PORT as string) }, resolve),
-  );
+  const port = parseInt(SERVER_PORT || "4000", 10);
```
Comment (beingnoble03) [inline src/index.ts:128]: let's keep this variable with upper case convention `PORT`

**Review by beingnoble03 — APPROVED** [review]
lgtm, just a nitpick

**Comment by Suyash878** [comment]
@palisadoes To be clear do you mean like the .env file should not have the parameter in itself completely?  

**Suggested Screenshot:**
image

So are you asking if my logic works for this case or not?

**Comment by beingnoble03** [comment]
Yes @Suyash878, does this work in that case?


Also, please fix the failing test. `{port: PORT}` should work there i think.

**Comment by Suyash878** [comment]
@beingnoble03 Thank you for the suggestion!

**Comment by Suyash878** [comment]
@palisadoes Yes it does work for that case.

Here's a video for better visualization.
[Screencast from 2024-07-16 22-04-38.webm](https://github.com/user-attachments/assets/beaf4475-e1e0-43bb-bbf2-3cbda60fdfff)

**Review by beingnoble03 — CHANGES_REQUESTED** [review]
@Suyash878 run prettier to fix the lint failure.

**Review by beingnoble03 — APPROVED** [review]
